superset-frontend/packages/superset-ui-core/src/query/getClientErrorObject.ts

```diff
@@ -171,9 +197,14 @@ export function getClientErrorObject(
     // fall back to Response.statusText or generic error of we cannot read the response
     let error = (response as any).statusText || (response as any).message;
     if (!error) {
-      // eslint-disable-next-line no-console
-      console.error('non-standard error:', response);
-      error = t('An error occurred');
+      // check response http status code and return a generic error message
+      const { status } = response as any;
```

Comment:
So it seems this only gets triggered only if the response object has no `statusText` and no `message`(?) Yet I think we've established that the "message" exist but also may be junk or complex html we don't want to surface as is. 

I'm not sure if that's best. I'd say if 1) there is proper response object, and 2) that response object has an error code and [maybe] 3) we haven't identified a proper/custom/specific error message, we should roll with the http error code as the way to define the message.

Thinking about precedence here:

1. ideally we have a json blob with a clear error message. This is best case and we want to surface that error message that came in the payload. Best case.
2. we have a proper http response with an error code. Here I'd say if `parseErrorString` returns a clear hit then use that, and if not fall back on `ERROR_CODE_LOOKUP` (?). Here I could also go with a pure ERROR_CODE_LOOKUP over `parseErrorString`
3. if we received a string (not a response object) we just have a string and no response object, here maybe we do the `parseErrorString`

I don't like that polymorphic function that is set to receive one of 3 types. Not sure if the refactor belongs in this PR, but I'd like to have consistency on what this function receive. Why can't it always get a response object and never a string? Would it make sense to see all the calls and make them more uniform?

Reply:
Tried to refactor to suit the order of ops better. If the JSON message is HTML it looks for the status code first then goes through the HTML code to try to pull the error, otherwise surfaces the good message, else it treats the body as text and then looks for the status code or tries the html parsing, last resort it shows errorText.